src/record.rs

```diff
@@ -79,41 +79,41 @@ impl MSSampleType {
 }
 
 /// An enumeration of possible data encodings.
-#[repr(i8)]
+#[repr(u8)]
 #[derive(Debug, Eq, PartialEq, Copy, Clone)]
 pub enum MSDataEncoding {
     /// Text encoding (UTF-8)
-    Text = raw::DE_TEXT as c_char,
+    Text = raw::DE_TEXT as u8,
```

Comment:
Sorry if this is a dumb rust newby question.   Is this cast needed?

Above `MSDataEncoding` is now specified as `u8`, so without this cast I would expect an error if `DE_TEXT` can't be expressed as a `u8`.  Based on the code below, I believe this casts `DE_TEXT` from a `ui32` to a `u8`.  If the `DE_TEXT` value can't be expressed as `u8`, does this cast mask the error?

Reply:
In here, we need to cast to `u8` instead of `as _` because we need to convert this value to string. If we use `as _` then we couldn't determine which api should use to convert this value to string, i.e, either `i8_to_string` or `to_sting`.

You are right, casting should handle it crefully. This literals are created by bindgen crate their types are u32, but the actual value of this literals are < 255 and casting to u8 shouldn't be a problem.

This casting inside `MSDataEncoding` is safe and doesn't prvide any invalid value as the value of all literals within the limit of `u8`

Reply:
> we need to convert this value to string

I'm confused, where does "string" come in?  Doesn't this cast `raw::DE_TEXT` which is a literal `0` to a `enum MSDataEncoding` which is to be encoded as a `u8`?